contributors/design-proposals/job.md

```diff
@@ -26,6 +27,31 @@ Jobs are needed for executing multi-pod computation to completion; a good exampl
 here would be the ability to implement any type of batch oriented tasks.
 
 
+## Backoff policy and failed pod limit
+
+By design, Jobs do not have any notion of failure, other than Pod's `restartPolicy`
+which is mistakenly taken as Job's restart policy ([#30243](https://github.com/kubernetes/kubernetes/issues/30243),
+[#[43964](https://github.com/kubernetes/kubernetes/issues/43964)]).  There are
+situation where one wants to fail a Job after some amount of retries over certain
+period of time, due to a logical error in configuration etc.  To do so we are going
+following fields will be introduced, which will control the exponential backoff
+when retrying Job: number of retries and time to retry.  The two fields will allow
+creating a fine grain control over the backoff policy, limiting the number of retries
+over specified period of time.  In the case when only one of them is specified
+an exponential backoff with duration of 10 seconds and factor of 2 will be applied
+in such a way that either time or number is reached.  After reaching the limit
```

Comment:
When creating exponential backoff with only number, I'll use the number as how many retries should be performed (using 10s as the time between retries, with factor of 2) see [Backoff and ExponentialBackoff](https://github.com/kubernetes/kubernetes/blob/e917f4ed39967e02edcd64d0574f22edc4d7e76f/staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go#L139-L172) for more info about these. In the case with time, I'll use the time as the max duration of the backoff. Does it make sense now?